.github/workflows/test.yml

```diff
@@ -1,38 +1,49 @@
-on: push
+on:
+  push:
+  pull_request:
+    types: [synchronize]
 name: Test
 jobs:
   test:
     strategy:
       fail-fast: false
       matrix:
-        go-version: [1.16.x, 1.17.x]
+        go-version: [1.17.x, 1.18.x]
         platform: [ubuntu-latest, macos-latest, windows-latest]
     runs-on: ${{ matrix.platform }}
     steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
       - name: Install Go
-        uses: actions/setup-go@v1
+        uses: actions/setup-go@v2
         with:
           go-version: ${{ matrix.go-version }}
-      - name: Install staticcheck
-        run: go install honnef.co/go/tools/cmd/staticcheck@latest
-        shell: bash
-      - name: Install golint
-        run: go install golang.org/x/lint/golint@latest
-        shell: bash
-      - name: Update PATH
-        run: echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
-        shell: bash
-      - name: Checkout code
-        uses: actions/checkout@v1
       - name: Fmt
         if: matrix.platform != 'windows-latest' # :(
         run: "diff <(gofmt -d .) <(printf '')"
         shell: bash
       - name: Vet
         run: go vet ./...
       - name: Staticcheck
-        run: staticcheck ./...
-      - name: Lint
-        run: golint ./...
+        uses: dominikh/staticcheck-action@v1
+        with:
+          install-go: false
+          cache-key: ${{ matrix.go-version }}
       - name: Test
         run: go test -race ./...
+      - name: Test coverage
+        run: go test -coverprofile="cover.out" ./... # quotes needed for powershell
+      - name: Send coverage
+        uses: shogo82148/actions-goveralls@v1
+        with:
+          path-to-profile: cover.out
+          flag-name: go${{ matrix.go-version }}-${{ matrix.os }}
+          parallel: true
```

Comment:
I took this out of the main repo, I think, at some point, due to flakiness. Do you have more confidence in it now?

Reply:
I copied this from `github.com/go-logfmt/logfmt` where it's been working fine for many months. I have that repo set to rerun the actions every 10 days even without any commits and haven't had any issues.